### PR TITLE
Add function type for `popperConfig` option

### PR DESCRIPTION
**Pull request by rohit2sharma95**
Since shallow merge is used to merge `popperConfig` passed by the user for popper's config, this overrides the `modifiers` property instead of merging the array.

So creating `popperConfig` a function and then passing the default configuration as an argument gives users the ability to use the default configuration and merge it with their configuration. Keeping the previous type `object` in the supported types since removing it would be a breaking change.

Ref: https://github.com/twbs/bootstrap/pull/32443#issuecomment-748806985

@XhmikosR Let me know if it is a good idea :slightly_smiling_face:

### CODE REVIEW

**Comment by XhmikosR** [comment]
@rohit2sharma95 can you please add a test?

File: site/content/docs/5.0/components/dropdowns.md [hunk 1]
```diff
@@ -908,9 +908,9 @@ Options can be passed via data attributes or JavaScript. For data attributes, ap
     </tr>
     <tr>
       <td><code>popperConfig</code></td>
-      <td>null | object</td>
+      <td>null | object | function</td>
       <td><code>null</code></td>
-      <td>To change Bootstrap's default Popper config, see <a href="https://popper.js.org/docs/v2/constructors/#options">Popper's configuration</a></td>
+      <td>To change Bootstrap's default Popper config, see <a href="https://popper.js.org/docs/v2/constructors/#options">Popper's configuration</a>. If you pass the <code>popperConfig</code> as a function, it accepts the Bootstrap's default Popper config as an argument, you need to return the configuration from this function as an object.</td>
```
Comment (XhmikosR) [inline site/content/docs/5.0/components/dropdowns.md:913]: I'm not sure this is self-explanatory. Can we maybe reword this and all the instances in this PR? Perhaps with an example?

Comment (mdo) [reply site/content/docs/5.0/components/dropdowns.md:913]: Agreed on wording here, needs a rev. After that should be good to go from my perspective.

Comment (rohit2sharma95) [reply site/content/docs/5.0/components/dropdowns.md]: Updated :+1:

**Comment by XhmikosR** [comment]
@mdo please check the wording and maybe if the example is easy to grasp. It might need some further tweaking :)

**Review by mdo — APPROVED** [review]
Pushed another rev of copy changes here, should be good to go from my perspective.
